rslib/src/notetype/fields.rs

```diff
@@ -35,18 +35,22 @@ impl NoteField {
         }
     }
 
-    pub(crate) fn fix_name(&mut self) {
-        // remove special characters
+    pub(crate) fn fix_name(&mut self) -> bool {
+        // Return whether the name is valid. Remove special characters if it leads to a valid name.
         let bad_chars = |c| c == ':' || c == '{' || c == '}' || c == '"';
         if self.name.contains(bad_chars) {
             self.name = self.name.replace(bad_chars, "");
         }
         // and leading/trailing whitespace and special chars
         let bad_start_chars = |c: char| c == '#' || c == '/' || c == '^' || c.is_whitespace();
         let trimmed = self.name.trim().trim_start_matches(bad_start_chars);
+        if trimmed.is_empty() {
+            return false;
+        }
```

Comment:
It would be nicer to have fix_name return a Result<()> instead of a bool, as it's less surprising that the mutation doesn't happen when an error is returned.

Reply:
Thanks.

I'm really new to Rust, learned it for Anki. Nice to get feedback about the proper way. I'll correct it soon.

Reply:
Done. And that also allow to use try_for_each, which is nice here